diff --git a/sysc/init b/sysc/init
index 915685b..1f0d945 100755
--- a/sysc/init
+++ b/sysc/init
@@ -37,12 +37,12 @@ install_tar sed-4.0.9 0
 
 install_tar autoconf-2.64 0
 install_tar autoconf-2.69 0
+install_tar automake-1.10.3 0
 install_tar automake-1.11.2 0
 install_tar binutils-2.24 0
 install_tar bzip2-1.0.8 0
 install_tar bison-3.4.1 2
 install_tar coreutils-6.10 0
-install_tar curl-7.83.0 0
 install_tar dhcpcd-9.4.1 0
 install_tar diffutils-2.7 0
 install_tar findutils-4.2.33 0
@@ -55,7 +55,7 @@ install_tar libtool-2.2.4 0
 install_tar linux-headers-5.10.41 0
 install_tar m4-1.4.7 0
 install_tar make-3.82 0
-install_tar musl-1.2.3 0
+install_tar musl-1.2.3 1
 install_tar perl-5.6.2 0
 install_tar util-linux-2.19.1 0
 
