diff --git a/sysc/run.sh b/sysc/run.sh
index e9da3f4..55be06c 100755
--- a/sysc/run.sh
+++ b/sysc/run.sh
@@ -34,6 +34,14 @@ populate_device_nodes
 
 create_fhs
 
+if [ -e "${SOURCES}/distfiles" ]; then
+    mv "${SOURCES}/distfiles" /
+else
+    mkdir -p "${DISTFILES}"
+fi
+
+build curl-7.83.0 curl-7.83.0-pass1.sh
+
 # Obtain network connection
 if [ "${CHROOT}" = "False" ]; then
     dhcpcd --waitip=4
@@ -50,12 +58,6 @@ if [ "${CHROOT}" = "False" ]; then
     done
 fi
 
-if [ -e "${SOURCES}/distfiles" ]; then
-    mv "${SOURCES}/distfiles" /
-else
-    mkdir -p "${DISTFILES}"
-fi
-
 build bash-5.1
 
 exec env -i PATH="${PATH}" HOME="${HOME}" SOURCE_DATE_EPOCH="${SOURCE_DATE_EPOCH}" bash run2.sh
