diff --git a/sysa/musl-1.2.3/musl-1.2.3.sh b/sysa/musl-1.2.3/musl-1.2.3.sh
index 330b75a..0640cf0 100755
--- a/sysa/musl-1.2.3/musl-1.2.3.sh
+++ b/sysa/musl-1.2.3/musl-1.2.3.sh
@@ -2,6 +2,17 @@
 #
 # SPDX-License-Identifier: GPL-3.0-or-later
 
+src_get() {
+    # Before the last pass the tar file is moved before
+    # the linux build removes all distfiles to save space.
+    if [ -e "../${pkg}.tar.gz" ]; then
+        mkdir "${DISTFILES}"
+        mv "../${pkg}.tar.gz" "${DISTFILES}"
+    else
+        default
+    fi
+}
+
 src_configure() {
     CC=gcc ./configure \
         --host=i386-unknown-linux-musl \
